app/models/custom_button.rb

```diff
@@ -143,4 +146,19 @@ def self.get_user(user)
     raise _("Unable to find user '%{user}'") % {:user => user} if user.nil?
     user
   end
+
+  def clone(options = {})
+    options[:guid] ||= MiqUUID.new_guid
```

Comment:
I do not see a use case for a user supplied guid here.  We should make it internal for now and expose that option if we find we need it.